src/Symfony/Component/Yaml/Parser.php

```diff
@@ -1158,7 +1158,18 @@ private function lexInlineQuotedString(int &$cursor = 0): string
     private function lexUnquotedString(int &$cursor): string
     {
         $offset = $cursor;
-        $cursor += strcspn($this->currentLine, '[]{},:', $cursor);
+
+        while ($cursor < strlen($this->currentLine)) {
+            if (in_array($this->currentLine[$cursor], ['[', ']', '{', '}', ',', ':'], true)) {
+                break;
+            }
+
+            if (' ' === $this->currentLine[$cursor] && '#' === ($this->currentLine[$cursor + 1] ?? '')) {
```

Comment:
Shouldn't we handle `\t` as well?

Reply:
According to https://yaml.org/spec/1.2.0/#id2577376, tabs are indeed valid as separate spaces too.

Reply:
Indeed, added as well.